app/web/src/components/IdeHeader/IdeHeader.tsx

```diff
@@ -126,10 +143,41 @@ export default IdeHeader
 function DefaultTopButtons({
   project,
   projectTitle,
+  projectId,
   _projectOwner,
   handleRender,
   canEdit,
 }) {
+  const { currentUser } = useAuth()
+  const { pathname } = useLocation()
+
+  const [createFork] = useMutation(FORK_PROJECT_MUTATION, {
+    onCompleted: ({ forkProject }) => {
+      const params = {
+        userName: forkProject?.user?.userName,
+        projectTitle: forkProject?.title,
+      }
+      navigate(
+        pathname.includes('/ide') ? routes.ide(params) : routes.project(params)
+      )
```

Comment:
This seems a little brittle to me, as I would hope that the component would be able be able to know what page it's on without having to look at the route. What if someone names their project "idea" than it would match on "user/idea" I had a look at seeing what it would take to pass this information to the component instead and realised that it could be done but it would be adding more mess to the messy situation that is this component so I did a big refactor aimed at this branch.

I've put you as a review, but the diff is not super meaningful as it's really just huge chunks of code moved around 🤷 

https://github.com/Irev-Dev/cadhub/pull/534